src/System.Device.Gpio/System/Device/Gpio/PinValuePair.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Device.Gpio
+{
+    /// <summary>
+    /// Simple wrapper for a pin/value pair
+    /// </summary>
+    public readonly struct PinValuePair
```

Comment:
I'm not yet convinced that this adds enough value, and even if we decide adding it, then I would suggest changing a lot of our public api that takes in a pin and a value to use this instead. Because of that, I would separate this change to a different proposal/PR.

Reply:
> if we decide adding it, then I would suggest changing a lot of our public api that takes in a pin and a value to use this instead

Is it really worth forcing constructing another struct for APIs that only ever take one pin?

Reply:
I'm not sure, I'm trying to think on scenarios where you would get a collection of these and then you wanted to just take one and set the value, so it might be handy to just be able to pass this to the Write method as is. In any case, my point was not that we won't want this in the future, just that I think that we may want to think about it a bit more and perhaps discuss this with an Api Review panel, whereas the main change in this PR (the addition of the interface) we definitely know we want, which is why I was suggesting separating it.

Reply:
I think it is fair to say that we definitely want a way to pass sets of pins without allocating. You can't do that with ValueTuple (you can pass one, not multiple as you'd have to allocate an array). How we structure the api around the pairs and how we name is totally debatable and requires thought, but I think it is unlikely that we'd want to make it hard to not allocate.

The only other option is to pass individual Spans for pins and values, which I think is a non-starter.

Reply:
It does make it harder for me to write against the new APIs if we keep this a ValueTuple, which is an additional reason I don't want to leave this as is. I'm literally writing additional expander device drivers as we speak. I'm pretty convinced that we wouldn't accept a ValueTuple in design review. :)